src/Middleware/SubFolderMiddleware.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Yiisoft\Router\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Yiisoft\Router\Exception\BadUriPrefixException;
+use Yiisoft\Router\UrlGeneratorInterface;
+
+/**
+ * This middleware supports routing when the project is placed in a subfolder relative to webroot
+ */
+class SubFolderMiddleware implements MiddlewareInterface
```

Comment:
1. Does it make sense to merge it with primary routing middleware? Are there cases when it is not desired to have it?
2. Is the class meant to be extended?